From: =?utf-8?b?SmFuIMSMZXJuw70=?= <jcerny@redhat.com>
Date: Tue, 24 May 2022 12:15:44 +0200
Subject: Use correct includes

rpmvercmp is defined in rpm/rpmver.h
risdigit is defined in rpm/rpmstring.h

Resolves: rhbz#2080210
---
 cmake/FindRPM.cmake                    | 3 +++
 config.h.in                            | 1 +
 src/OVAL/results/oval_cmp_evr_string.c | 5 +++++
 3 files changed, 9 insertions(+)

diff --git a/cmake/FindRPM.cmake b/cmake/FindRPM.cmake
index a666942..369d153 100644
--- a/cmake/FindRPM.cmake
+++ b/cmake/FindRPM.cmake
@@ -30,6 +30,9 @@ set(RPM_VERSION ${RPM_PKGCONF_VERSION})
 if(RPM_VERSION)
 	string(COMPARE GREATER "4.6" ${RPM_VERSION} RPM46_FOUND)
 	string(COMPARE GREATER "4.7" ${RPM_VERSION} RPM47_FOUND)
+	if(NOT (RPM_VERSION VERSION_LESS "4.18"))
+		set(RPM418_FOUND 1)
+	endif()
 endif()
 
 # Set the include dir variables and the libraries and let libfind_process do the rest.
diff --git a/config.h.in b/config.h.in
index 1b72855..bb1428a 100644
--- a/config.h.in
+++ b/config.h.in
@@ -44,6 +44,7 @@
 #cmakedefine HAVE_RPMVERCMP
 #cmakedefine RPM46_FOUND
 #cmakedefine RPM47_FOUND
+#cmakedefine RPM418_FOUND
 
 #cmakedefine BZIP2_FOUND
 
diff --git a/src/OVAL/results/oval_cmp_evr_string.c b/src/OVAL/results/oval_cmp_evr_string.c
index 3bfc8ce..3ba0fa0 100644
--- a/src/OVAL/results/oval_cmp_evr_string.c
+++ b/src/OVAL/results/oval_cmp_evr_string.c
@@ -37,7 +37,12 @@
 #include "common/_error.h"
 
 #ifdef HAVE_RPMVERCMP
+#ifdef RPM418_FOUND
+#include <rpm/rpmver.h>
+#include <rpm/rpmstring.h>
+#else
 #include <rpm/rpmlib.h>
+#endif
 #else
 #ifdef OS_WINDOWS
 #include <malloc.h>
